### PR TITLE
KAFKA 8311 : Augment the error message to let user change `default.api.timeout.ms`

### PR DESCRIPTION
**Pull request by clearpal7**
The goal of this commit is from :
https://issues.apache.org/jira/browse/KAFKA-8311

1.  It is required to suggest a more meaningful message when `default.api.timeout.ms` that Timeout Exception could have occurred.
The 'default.api.timeout.ms' option coverage that was referenced by https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=75974886

2. It also needs to provide what stream thread was occurred of time out exception.
so I augment the timeout Exception message on Stream Thread which includes Restore Consumer and Thread Consumer in run() method on Stream Thread Class.

I`m wondering that the KAFKA 8311 merge request ticket is already written by  abbccdda 
on https://github.com/apache/kafka/pull/6662
 
I don`t know when the commit ticket was merged so I tried to write this merge request.
If this ticket is not appropriate on KAFKA policy. please comment on reply.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 1]
```diff
@@ -773,6 +774,10 @@ public void run() {
         try {
             runLoop();
             cleanRun = true;
+        } catch (final TimeoutException e) {
+
```
Comment (abbccdda) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: nit: remove this extra line

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 2]
```diff
@@ -773,6 +774,10 @@ public void run() {
...
+            throw new TimeoutException("The thread-consumer : " + threadMetadata.consumerClientId() + " and restore-consumer : " + threadMetadata.restoreConsumerClientId() + "has happened TimeoutExceptio" +
```
Comment (abbccdda) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: typo in the end: `has happened TimeoutExceptio`

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java [hunk 3]
```diff
@@ -1365,7 +1365,7 @@ public void commitSync(Duration timeout) {
             maybeThrowInvalidGroupIdException();
             if (!coordinator.commitOffsetsSync(subscriptions.allConsumed(), time.timer(timeout))) {
                 throw new TimeoutException("Timeout of " + timeout.toMillis() + "ms expired before successfully " +
-                        "committing the current consumed offsets");
+                        "committing the current consumed offsets. Try tuning 'default.api.timeout.ms' option");
```
Comment (abbccdda) [inline clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: I feel a better instruction needs a tuning direction, maybe `Try increasing 'default.api.timeout.ms' as appropriate`

**Review by abbccdda — COMMENTED** [review]
Some minor comments. Will let @mjsax @vvcephei also take a look to see if the error log change makes sense.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 1 above)
Comment (clearpal7) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: I removed the extra line.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java (hunk 2 above)
Comment (clearpal7) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: I modified the TimeoutException

File: clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java (hunk 3 above)
Comment (clearpal7) [reply clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java]: yep, the direction is a little obscure, so I modified the direction.

**Comment by mjsax** [comment]
@clearpal7 @abbccdda Just cleaning up my old backlog. Do you still want to get this in? PR would need a rebase.

**Comment by abbccdda** [comment]
Closing the PR due to lack of interest and duplicate.

**Pull request closed without merging**